File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  const trimmed = message.trim();
  
  if (trimmed === '') {
    return 'Fine. Be that way!';
  }

  const isQuestion = trimmed.endsWith('?');
  const isYelling = /[a-z]/i.test(trimmed) && trimmed === trimmed.toUpperCase();

  if (isYelling && isQuestion) {
    return "Calm down, I know what I'm doing!";
  }

  if (isYelling) {
    return 'Whoa, chill out!';
  }

  if (isQuestion) {
    return 'Sure.';
  }

  return 'Whatever.';
}

File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val remark = input.trim()
        
        if (remark.isEmpty()) {
            return "Fine. Be that way!"
        }

        val isQuestion = remark.endsWith("?")
        val hasLetters = remark.any { it.isLetter() }
        val isYelling = hasLetters && remark == remark.uppercase()

        return when {
            isYelling && isQuestion -> "Calm down, I know what I'm doing!"
            isYelling -> "Whoa, chill out!"
            isQuestion -> "Sure."
            else -> "Whatever."
        }
    }
}
